homeassistant/components/media_player/panasonic_viera.py

```diff
@@ -56,8 +56,13 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
         name = discovery_info.get('name')
         host = discovery_info.get('host')
         port = discovery_info.get('port')
+        udn = discovery_info.get('udn')
+        if udn and udn.startswith('uuid:'):
+            uuid = udn[len('uuid:'):]
```

Comment:
Is there some reason to snip this away, considering it's part of the value?

Reply:
I think it's redundant, and it doesn't contribute to the uniqueness. If there is a scenario where sometimes the value is returned with the prefix and sometimes not, perhaps between different TV models (or perhaps even using different protocols for getting the uuid), it's better to be consistent and just use the value. Just speculating here though, I've only tested with one device and don't know if the scenario described exists. Wouldn't hurt to prepare for it nevertheless IMO.

Reply:
agree with @scop 

it's completely redundant